variables.tf

```
@@ -1,10 +1,21 @@
-variable "saml_provider_name" {
-  description = "Name of the SAML Identity Provider"
-  type        = string
-}
-
-variable "saml_provider_metadata" {
-  description = "XML metadata for the SAML Identity Provider"
-  default     = ""
-  type        = string
+variable "iam_identity_provider" {
+  description = "Inputs to create a SAML Provider or OpenID Provider"
+  type = object({
+    saml = optional(object({
+      metadata_document = string
+      provider_name     = string
+    }))
+    openid_connect = optional(object({
+      client_id_list  = list(string)
+      thumbprint_list = optional(list(string), [])
```

The default for this should probably remain null, to represent unset. An empty list would imply it should be set to zero items. Often the provider or the API will interpret those two conditions differently.

```suggestion
      thumbprint_list = optional(list(string))
```

---

Removed empty list